doc/src/site/sphinx/tutorials/spark_h2o_conversions.rst

```diff
@@ -11,108 +11,85 @@ Quick links:
+The call expects the type ``A`` to create a correctly typed RDD. The conversion requires type ``A`` to be bound by the ``Product`` interface.
+The relationship between the columns of the H2OFrame and the attributes of class ``A`` is based on name matching.
 
 Example
 ^^^^^^^
 
 .. code:: scala
 
-    val df: H2OFrame = ...
-    val rdd = asRDD[Weather](df)
-
---------------
+    case class Person(name: String, age: Int)
+    val rdd = asRDD[Person](h2oFrame)
 
 Converting an H2OFrame into a DataFrame
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``H2OContext`` class provides the explicit conversion, ``asDataFrame``, which creates a DataFrame-like wrapper around the provided H2O H2OFrame. Technically, it provides the ``RDD[sql.Row]`` RDD API:
+The ``H2OContext`` class provides the method ``asDataFrame``, which creates a DataFrame-like wrapper around the provided H2OFrame:
 
 .. code:: scala
 
-    def asDataFrame(fr : H2OFrame): DataFrame
-
-This call does not require any type of parameters, but because it creates ``DataFrame`` instances, it requires access to an instance of ``SQLContext``. In this case, the instance is provided as an implicit parameter of the call. The parameter can be passed in two ways: as an explicit parameter or by introducing an implicit variable into the current context.
+    def asDataFrame(fr: H2OFrame): DataFrame
 
-The schema of the created instance of the ``DataFrame`` is derived from the column name and the type of ``H2OFrame`` specified.
+The schema of the created instance of the ``DataFrame`` is derived from the column names and types of the specified ``H2OFrame``.
 
 Example
 ^^^^^^^
 
-Using an explicit parameter in the call to pass sqlContext:
-
 .. code:: scala
 
-    val sqlContext = new SQLContext(sc)
-    val schemaRDD = asDataFrame(h2oFrame)
-
-or as an implicit variable provided by the actual environment:
-
-.. code:: scala
-
-    implicit val sqlContext = new SQLContext(sc)
-    val schemaRDD = asDataFrame(h2oFrame)
-
---------------
+    val dataFrame = asDataFrame(h2oFrame)
 
 Converting an RDD[T] into an H2OFrame
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The ``H2OContext`` provides **implicit** conversion from the specified ``RDD[A]`` to ``H2OFrame``. As with conversion in the opposite direction, the type ``A`` has to satisfy the upper bound expressed by the type ``Product``. The conversion will create a new ``H2OFrame``, transfer data from the specified RDD, and save it to the H2O K/V data store.
+The ``H2OContext`` provides conversion method from the specified ``RDD[A]`` to ``H2OFrame``. As with conversion
+in the opposite direction, the type ``A`` has to satisfy the upper bound expressed by the type ``Product``. The conversion
+creates a new ``H2OFrame``, transfers data from the specified RDD, and saves it to to the H2O backend.
```

Comment:
Couldn't `H2O backend` be misinterpreted with the external/internal backend?

Reply:
Good catch -> reworded to `to DKV store on the H2O backend.`, WDYT?

Reply:
Sounds good to me